sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeed.cs

```diff
@@ -245,18 +206,25 @@ internal DataFeedDetail GetDataFeedDetail()
         /// </summary>
         internal DataFeedDetailPatch GetPatchModel()
         {
-            DataFeedDetailPatch patch = DataSource.InstantiateDataFeedDetailPatch();
+            DataFeedDetailPatch patch = DataSource?.InstantiateDataFeedDetailPatch()
+                ?? new DataFeedDetailPatch();
 
             patch.DataFeedName = Name;
             patch.Status = Status.HasValue ? new DataFeedDetailPatchStatus(Status.ToString()) : default(DataFeedDetailPatchStatus?);
 
-            patch.TimestampColumn = Schema.TimestampColumn;
+            if (Schema != null)
+            {
+                patch.TimestampColumn = Schema.TimestampColumn;
+            }
 
-            patch.DataStartFrom = ClientCommon.NormalizeDateTimeOffset(IngestionSettings.IngestionStartTime);
-            patch.MaxConcurrency = IngestionSettings.DataSourceRequestConcurrency;
-            patch.MinRetryIntervalInSeconds = (long?)IngestionSettings.IngestionRetryDelay?.TotalSeconds;
-            patch.StartOffsetInSeconds = (long?)IngestionSettings.IngestionStartOffset?.TotalSeconds;
-            patch.StopRetryAfterInSeconds = (long?)IngestionSettings.StopRetryAfter?.TotalSeconds;
+            if (IngestionSettings != null)
+            {
+                patch.DataStartFrom = IngestionSettings.IngestionStartTime.HasValue ? ClientCommon.NormalizeDateTimeOffset(IngestionSettings.IngestionStartTime.Value) : null;
```

Comment:
so the service will still work if `IngestionStartTime  == null` ? was this a misunderstanding on our side? or what changed?

Reply:
The `IngestionStartTime` is required and its presence is enforced during Data Feed creation. This piece of code is called during the `UpdateDataFeed` method. If `IngestionStartTime` is null here, it simply means that we don't want to update it. Before we made it nullable, whenever the user wanted to update the `IngestionSettings` of a Data Feed, they were forced to specify the `IngestionStartTime` again, since it was not nullable and required in the constructor.